flipperdevices/flipperzero-firmware
# [no_ticket] Bandwidth, Datarate and Macnhester Enable decoding configuration

# What's new

- [SubGhz] 
   -  [Receiver]
      Special preset `AM*` introduced. This preset allows to modify `Bandwidth`, `Datarate` and `Macnhester decoding` parameters of `CC1101` chip.
      This PR also allows to get information about these fields from default (or other custom) modulation presets
      
- [unit_tests]
  - Minor improvement to unit_tests app - allows to set desired log level before running unit_tests

## Abstract
Some transmitters may have non-standard transmission params. So in order to get pulses more accurate you might need to have options to adjust some `CC1101` settings manually.

This PR also could be a basement for other configurable parameters of `CC1101` chip editing.

# Verification 

- Open SubGHz App and enter *read raw* mode:
  ![0000](https://user-images.githubusercontent.com/5215122/210118045-63c6108a-adf0-4aaa-81ba-455ed7b6783c.png)
- Select modulation preset `AM*`
  ![0001](https://user-images.githubusercontent.com/5215122/210118048-148e6679-c630-4ec4-895a-61a44a8ddc88.png)
  ![0002](https://user-images.githubusercontent.com/5215122/210118047-5ebe3866-3c91-4b1f-b791-14fd3a2660ba.png)
  > Please note: default preset `AM650` will show invalid `Bandwidth`, `Datarate` and `Manchester decoding` params because this preset was initialized with `NULL` as it `CC1101`'s registers data
- Setup desired configuration and test it
  ![0003](https://user-images.githubusercontent.com/5215122/210118046-12733715-d1c6-4f05-8da9-64934903a57b.png) 
- Recorded RAW file will contain `CC1101` registers configuration according to your setup
- Loading this RAW file should setup exact configuration
  > Please note: `CUSTOM` presets also allowed to be modified, so once you load your RAW file, you may change the advanced configuration of custom preset for RAW file

# Checklist (For Reviewer)
